File: examples/python/shared/src/lemma_examples_shared/tools.py
from .docs import list_docs, read_doc


def execute_docs_tool(name: str, args: dict[str, object] | None = None) -> str:
    payload = args or {}
    if name == "list_docs":
        return list_docs()
    if name == "read_doc":
        url = payload.get("url")
        if not isinstance(url, str) or not url.strip():
            raise ValueError("read_doc requires a string url argument")
        return read_doc(url)
    raise ValueError(f"Unknown tool: {name}")

File: examples/shared/src/tools.ts
import { listDocs, readDoc } from "./docs";

export async function executeDocsTool(
  name: string,
  args: Record<string, unknown> = {},
): Promise<string> {
  if (name === "list_docs") {
    return listDocs();
  }
  if (name === "read_doc") {
    const url = args.url;
    if (typeof url !== "string" || !url.trim()) {
      throw new Error("read_doc requires a string url argument");
    }
    return readDoc(url);
  }
  throw new Error(`Unknown tool: ${name}`);
}
